### ISSUE TITLE
about the hyperparametric gamma

### ISSUE THREAD

**Issue by qpy7777777** [unique-chan/Complement-Cross-Entropy#8]
the value of gamma is negative, but in the COT code, no negative number is displayed, because different from the standard cross entropy, the greater the supplementary entropy, the higher the flatness of the error category, so the COT author did not add a negative sign, the obtained supplementary entropy is negative, but the gamma value used by the author is negative. If softmax is # yHat = torch.tensor([[0.5,0.4,0.1]])
yHat = torch.tensor([[0.5,0.25,0.25]]), according to the code in this article, the standard cross entropy results are consistent, if 0.2, assume that the first compensation entropy loss is -0.25, and the second compensation entropy loss is -0.34. The second type has a smaller loss value and is consistent with subsequent loss minimization training. But when the author multiplies the negative gamma, the loss value of the second type will be larger, but isn't it hoped that the loss value of the first type will be larger?

**Comment by unique-chan** [comment]
To make the distribution flatten (<=> high uncertainty), we need to maximize the entropy of the distribution. One fact is that the PyTorch deep learning framework only has a minimizer (to minimize the training loss) as an optimizer. Thus, **we need to use negative gamma values** to maximize the entropy of the distribution of negative classes.

In the paper, the proposed loss function is defined as: **H(y, y_hat) + C(y_hat)**, where **H**: cross entropy, **C**: balanced complement entropy.

However, in our code, the loss function is defined as: **cross entropy - self.balancing_factor * complement_entropy.**

**Therefore, you do not need to add a negative sign for self.balancing_factor during training**. (e.g. if you want to gamma=-1 like paper, self.balancing_factor should be initialized with 1 (not -1)) Although this is to reduce the effort of adding '-' to each experiment, we realized that it might confuse users. Hope you understand.

Besides, sorry for the late reply.

**Comment by qpy7777777** [comment]
Thanks for your reply, I get it.
